Feature-PUT-shopId-#13

dbertozzi opened the pull request:
PUT /shops/:shopId route
Two separate routes - one for cover photo, and one for shop details
Edit the shop details + cover photo
S3 -> will use s3 to save photos, we will save the photo URL on the database
Note: shop is missing from the url. going to revisit tomorrow.
I did not find the video on slack soon enough and did not get to use multer. I already had it working with busboy. I can redo it if multer is something we should have used.

dbertozzi committed:
PUT shopid/image-upload and shopid/details

PUT /shops/:shopId route
Two separate routes - one for cover photo, and one for shop details
Edit the shop details + cover photo
S3 -> will use s3 to save photos, we will save the photo URL on the database

I did not find the video on slack soon enough and did not get to use multer. I already had it working with busboy. I can redo it if we need multer.

naapak commented on server/app.js:
```diff
@@ -4,36 +4,41 @@ import { join } from "path";
-app.use("/auth", auth);
 app.use("/", indexRouter);
+app.use("/ping", pingRouter);
+app.use("/auth", authRouter);
```
I see you changed the auth router end point. We should have a map of all the routes for both frontend and backend to be on the same page as the app gets bigger.

dbertozzi replied:
no change the endpoint URL. Just changed the import variable name to make it consistent with the other router objects.

naapak reviewed (approved):
Looks good to me. Good that you removed the .env from the git.

skassam21 commented on README.md:
```diff
@@ -9,6 +9,71 @@ To Run MongoDb in mac with mongoose, follow three steps
+6. Update the .env file with the following variables:
+BUCKET_NAME
+IAM_USER_KEY
+IAM_USER_SECRET
```
Good notes here 👍

skassam21 commented on server/app.js:
```diff
@@ -4,36 +4,41 @@ import { join } from "path";
-import auth from "./routes/auth";
-import verifyToken from "./routes/verifyToken";
+import authRouter from "./routes/auth";
+const shopIdRouter = require("./routes/shopId");
```
can be consistent with how you import? We should use the import statement as opposed to the older require statement

dbertozzi replied:
Do you have any resource to explain the difference? Every resource I have used for learning uses require for Node. This is the first time I have seen import used in Node (in React I use it regularly of course). I prefer consistency too, and almost changed everything to require.

skassam21 commented on server/app.js:
```diff
@@ -4,36 +4,41 @@ import { join } from "path";
-import auth from "./routes/auth";
-import verifyToken from "./routes/verifyToken";
+import authRouter from "./routes/auth";
+const shopIdRouter = require("./routes/shopId");
```
Also this is not a standard naming convention for the files with NodeJS - https://stackoverflow.com/questions/18927298/node-js-project-naming-conventions-for-files-folders
OR
https://google.github.io/styleguide/jsguide.html#file-name

Filenames should probably be kebab-case
I would change it to even just shop though to be honest

skassam21 commented on server/routes/shopId.js:
```diff
@@ -0,0 +1,56 @@
+const express = require("express");
+const router = express.Router();
+const AWS = require("aws-sdk");
+const verify = require('./verifyToken');
```
again, consistent importing

skassam21 commented on server/routes/shopId.js:
```diff
@@ -0,0 +1,56 @@
+    Bucket: process.env.BUCKET_NAME
+  });
+  let imageData;
+  s3bucket.createBucket(function() {
```
don't we need to just run this once? We shouldn't have to run this everytime we upload to S3?

skassam21 commented on server/routes/shopId.js:
```diff
@@ -0,0 +1,56 @@
+const AWS = require("aws-sdk");
+const verify = require('./verifyToken');
+
+function uploadToS3(file) {
```
I feel like this logic should go in another file, like a util or something

skassam21 commented on server/routes/shopId.js:
```diff
@@ -0,0 +1,56 @@
+    // Begins the upload to the AWS S3
+    const imageData = uploadToS3(file);
+    // to do: store image url in database
+    console.log(imageData);
```
Don't we have to do something with the imageData / URL ? Like save it to the Shop Model?

This looks good, but I feel like we'll have to work on these routes more to make them functional. This can probably happen in another PR though

dbertozzi replied:
I don't think the shop model has been created yet. Naresh has the ticket and is working on it, or perhaps finished by now.

skassam21 replied:
Sounds good

dbertozzi committed:
Used Multer instead of busboy

Updated to use import statement as requested. I still am not convinced because I have seen in recently made content using require server side scripts. Changed naming conventions of files and paths to use dashes instead of camelCase.

skassam21 commented on server/app.js:
```diff
@@ -4,22 +4,20 @@ import { join } from "path";
+  () => console.log('Mongoose has connected to the database!')
 );
-var db = mongoose.connection;
+let db = mongoose.connection;
```
probably const right? this doesn't change?

dbertozzi replied:
Good point.

skassam21 reviewed (approved):
One small point. We will have to start fixing the tests though (we getting those red x's)

naapak committed:
updating config file

naapak committed:
testing env variables

naapak committed:
wrong indentation fix test

dbertozzi committed:
Merge branch 'dev' into dbertozzi/feature-put-shopId-#13
